In [25]:
T.decode([13736])

'101'

In [32]:
T.unk_id()

2

In [33]:
T.id_to_piece(3)

'▁'

In [28]:
T.decode([3,11190])

'110'

In [4]:
!pip install jsonlines

  Using cached jsonlines-3.0.0-py3-none-any.whl (8.5 kB)


In [6]:
%load_ext autoreload
%reload_ext autoreload
%autoreload 2
from IPython.core.display import display, HTML
display(HTML("<style>.container { width:100% !important; }</style>"))
import json
import pandas as pd
import numpy as np
from tasks_with_tokens import *


tids = tuple(T5_TOKEN_IDS_FOR_TASKS)
def show_sample_examples(func, num_examples=10, length_range=(10, 20), token_ids=tids):
    for _ in range(num_examples):
        print(func(length_range=length_range, char_set=char_set))
        
UNARY_BASIC_TASKS = ['copy', 'reverse', 'set', 'first_char', 'last_char', 'deduplicate', 'length', 
       'longest_word', 'duplicate']

# UNARY_BASIC_TASKS = ['length']
T = TOKENIZER

def show_task_examples(task, is_natural_language=False, num_examples=20, is_print=True):
    for count in range(num_examples):
        basic_task_sampling_args = {'length_range': (10,30), 'token_ids':tids[:200]}
        example = generate_example(TASK_REGISTRY[task], is_natural_language=is_natural_language, **basic_task_sampling_args)
        if count % 1000==0:
            print(count)
        if is_print:
            print("-----")
            if task == 'longest_word':
                print(f'@input = {example[0]}')
                print(f'output = {example[1]}')
            else:
                print(example)
#             print(f"input token IDs {T.encode(example[0])}")
            print(f"input decoded: {T.decode(eval(example[0]))}")

            print(f"output decoded: {T.decode(eval(example[1]))}")

#             if len(example[1]) == 0 or example[1].isspace():
#                 print(example[1])
#                 raise RuntimeError()

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


/var/folders/xb/xhwwptys1xs5vn0fn_7c7_600000gn/T/ipykernel_9537/3995065402.py:4: DeprecationWarning: Importing display from IPython.core.display is deprecated since IPython 7.14, please import from IPython display
  from IPython.core.display import display, HTML


In [7]:
for task in UNARY_BASIC_TASKS:
    print(f'\n{task}')
    show_task_examples(task, num_examples=5, is_print=True)


copy
0
-----
('[2405, 32099, 129, 165, 148, 13, 132, 67, 2, 191, 106, 120, 73, 94, 102, 174, 50, 90, 183, 143, 57, 105, 134, 80, 3, 96, 36, 19, 173, 161, 191, 188]', '[129, 165, 148, 13, 132, 67, 2, 191, 106, 120, 73, 94, 102, 174, 50, 90, 183, 143, 57, 105, 134, 80, 3, 96, 36, 19, 173, 161, 191, 188]')
input decoded: copy <extra_id_0> get its You of there die ⁇  pentruonly un Itp need la le am make by “S one  " be isil work pentruA
output decoded: get its You of there die ⁇  pentruonly un Itp need la le am make by “S one  " be isil work pentruA
-----
('[2405, 32099, 62, 103, 94, 139, 178, 120, 162, 53, 69, 140, 43, 168, 11, 12, 51, 26, 82, 8, 55, 73, 39, 142, 42]', '[62, 103, 94, 139, 178, 120, 162, 53, 69, 140, 43, 168, 11, 12, 51, 26, 82, 8, 55, 73, 39, 142, 42]')
input decoded: copy <extra_id_0> we do It into uslyveing our me have well and tomd my the! un your se or
output decoded: we do It into uslyveing our me have well and tomd my the! un your se or
-----
('[2405, 32099, 194, 9

In [8]:
BINARY_BASIC_TASKS = ['count', 'delete', 'filter', 
       'get_index', 'search', 'sort', 
          'replace', 'replace_many', 'union', 'intersect', 'set_1_minus_2', 'set_2_minus_1']


for task in BINARY_BASIC_TASKS:
    print(f'\n{task}')
    show_task_examples(task, num_examples=5, is_print=True)


count
0
-----
('[3476, 32099, 21, 66, 72, 72, 126, 4, 72, 72, 72, 32097, 72]', '[305]')
input decoded: count <extra_id_0> for all more more newX more more more <extra_id_2> more
output decoded: 5
-----
('[3476, 32099, 199, 145, 145, 145, 145, 145, 145, 145, 186, 145, 99, 3, 145, 145, 145, 156, 145, 145, 145, 32097, 145]', '[968]')
input decoded: count <extra_id_0> help than than than than than than than many thanif  than than than If than than than <extra_id_2> than
output decoded: 14
-----
('[3476, 32099, 3, 191, 191, 191, 19, 183, 77, 54, 179, 195, 174, 191, 54, 191, 191, 191, 180, 191, 116, 191, 164, 191, 133, 191, 147, 32097, 191]', '[850]')
input decoded: count <extra_id_0>  pentru pentru pentru is amin canablell need pentru can pentru pentru pentru S pentru when pentru may pentru would pentru over <extra_id_2> pentru
output decoded: 11
-----
('[3476, 32099, 47, 47, 47, 47, 47, 47, 47, 47, 145, 47, 47, 32097, 47]', '[335]')
input decoded: count <extra_id_0> was was was was was wa

In [66]:
[1,2,2,3].index([2,2])

ValueError: [2, 2] is not in list

In [11]:
'a'.split(', ')

['a']

('last_char <extra_id_0> assessmentten Alonglapse', 'lapse')
input token IDs [336, 834, 4059, 32099, 4193, 324, 8529, 16543]
output token IDs [3, 16543]

In [35]:
T.id_to_piece(3)

'▁'

In [16]:
x = generate_set_data((4,20), TASKS_TOKEN_IDS)
print(x)

[8909, 2376, 4165, 26815, 26815, 4165, 26815, 30679, 30679, 4165, 2376, 4165, 8909, 4165, 8909]


In [23]:
print(TOKENIZER.encode(TOKENIZER.decode(x)))

[8909, 2376, 4165, 26815, 26815, 4165, 26815, 30679, 30679, 4165, 2376, 4165, 8909, 4165, 8909]
